File: Kotlin/Group_Anagrams.kt
class Solution {
    fun groupAnagrams(strs: Array<String>): List<List<String>> {
        val map = HashMap<String, MutableList<String>>()

        for (s in strs) {
            val count = IntArray(26)
            for (c in s) {
                count[c - 'a']++
            }

            val key = count.joinToString("#")
            if (!map.containsKey(key)) {
                map[key] = mutableListOf()
            }
            map[key]!!.add(s)
        }

        return map.values.toList()
    }
}

File: Cpp/Group_Anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> mp;
        for (string s : strs) {
            vector<int> count(26, 0);
            for (char c : s)
                count[c - 'a']++;
            string key;
            for (int num : count)
                key += to_string(num) + '#';
            mp[key].push_back(s);
        }
        vector<vector<string>> result;
        for (auto entry : mp)
            result.push_back(entry.second);
        return result;
    }
};